### PR TITLE
Added support for SASL PLAIN authentication

### PR DESCRIPTION
**Pull request by ppatierno**
This PR addresses #55

### CODE REVIEW

File: internal/config/canary_config.go [hunk 1]
```diff
@@ -160,10 +173,24 @@ func (c CanaryConfig) String() string {
...
+	SASLPassword := ""
+	if c.SASLMechanism == sarama.SASLTypePlaintext {
+		if c.SASLUser != "" {
+			SASLUser = "[SASL user]"
+		}
+
+		if c.SASLPassword != "" {
+			SASLPassword = "[SASL password]"
+		}
+	}
```
Comment (tombentley) [inline internal/config/canary_config.go:187]: If we just supporting SASL/PLAIN shouldn't we have an `else` here to error on other mechanisms? I can see you handle this in `auth.go`, but it seems cleaner to do it here, no?

Comment (ppatierno) [reply internal/config/canary_config.go:187]: this function is just for printing the configuration passed to the canary, it's kind of override of `toString` in Java :-) ... the if condition I am putting there is just for hiding the username/password with placeholders but nothing more.

Comment (tombentley) [reply internal/config/canary_config.go:187]: Ah, that's what I get for just scanning the diff. Thanks for the clarification.

**Review by tombentley — APPROVED** [review]

**Review by k-wall — APPROVED** [review]

File: internal/security/auth.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ func SetAuthConfig(canaryConfig *config.CanaryConfig, saramaConfig *sarama.Confi
 			return errors.New("SASL password must be specified")
 		}
 		saramaConfig.Net.SASL.Enable = true
+		saramaConfig.Net.SASL.Version = sarama.SASLHandshakeV1
```
Comment (k-wall) [inline internal/security/auth.go:28]: should that be configurable?

Comment (ppatierno) [reply internal/security/auth.go:28]: version V0 doesn't provide information about a SASL authentication failure while it's provided by V1 which is supported by ages. I don't see any value supporting V0 imho.

Comment (tombentley) [reply internal/security/auth.go:28]: See https://cwiki.apache.org/confluence/display/KAFKA/Kafka+APIs

Comment (tombentley) [reply internal/security/auth.go:28]: Supported since Kafka 1.0 in fact.

**Pull request merged**